valhalla/midgard/openlr.h

```diff
@@ -241,9 +294,12 @@ struct LineLocation {
     return result;
   }
 
-  LocationReferencePoint first;
-  LocationReferencePoint last;
-  std::vector<LocationReferencePoint> intermediate;
+  bool operator==(const LineLocation& lloc) const {
+    return lrps.size() == lloc.lrps.size() && poff == lloc.poff && noff == lloc.noff &&
+           std::equal(lrps.begin(), lrps.end(), lloc.lrps.begin());
+  }
+
+  std::vector<LocationReferencePoint> lrps;
```

Comment:
the bulk of the change in this file is due to me changing the record to a single vector of lrps instead of having first and last broken out